nemo_skills/code_execution/utils.py

```diff
@@ -16,16 +16,28 @@
 from typing import Dict, Tuple
 
 
-def format_code_output(execution_dict: Dict[str, str], code_output_begin: str, code_output_end: str):
+def format_code_output(
+    execution_dict: Dict[str, str], code_output_begin: str, code_output_end: str, code_output_format: str = 'llama'
+):
     """Formatting code output to be displayed as an llm expects it."""
-    output = execution_dict["process_status"]
-    if execution_dict['stdout']:
-        output += f"\n[stdout]\n{execution_dict['stdout']}\n[/stdout]"
-    if execution_dict['stderr']:
-        output += f"\n[stderr]\n{execution_dict['stderr']}\n[/stderr]"
+    if code_output_format == 'llama':
+        output = execution_dict["process_status"]
+        if execution_dict['stdout']:
+            output += f"\n[stdout]\n{execution_dict['stdout']}\n[/stdout]"
+        if execution_dict['stderr']:
+            output += f"\n[stderr]\n{execution_dict['stderr']}\n[/stderr]"
+        output = f"{code_output_begin}\n\n{output}{code_output_end}\n\n"
+    elif code_output_format == 'qwen':
+        output = ""
+        if execution_dict['stdout']:
+            output += f"{execution_dict['stdout']}\n"
```

Comment:
I don't think we should always be adding a new line, right? It's only needed if both stdout and stderr are present, but otherwise we just take whatever is returned. 

It's probably a minor detail, but it's good to get this as consistent as possible, since LLMs might get off-track if there is an extra newline somewhere where it's not supposed to be

Reply:
I guess an alternative is that we don't include newline and actually raise an assert if both stderr and stdout are non-empty. I think it should never be the case with our current sandbox and if it is, then we don't really know what to do, since stdout and stderr might have been interleaved in the output, but we extracted them separately

Reply:
> I don't think we should always be adding a new line, right? It's only needed if both stdout and stderr are present, but otherwise we just take whatever is returned.

Right now the output will look like
\`\`\`output
number
\`\`\`
if stdout equals to number. 

And if both stdout and stderr are empty we return 
\`\`\`output
\`\`\`
without any content.

Isn't it what we expect?

Reply:
>  I think it should never be the case with our current sandbox

What abut the code printing something before an error? Isn't this possible with our current sandbox?

Reply:
yes, but the issue is that if you have a bunch of prints to stderr and then to stdout and they are mixed. It's unclear how they should be treated. But it's not really a problem right now because for some reason (that's actually a bug but I don't know how to fix it) even errors go to stdout. E.g. you can see that in execution tests we always assert that stderr is empty

Reply:
But the newlines are already in the output, right? E.g. if you do print(1) it will be "1\n" and so your current change would make it "1\n\n"?

Reply:
Right, this is a good point